// otb-module-plus/src/components/hForm/interface.ts
import { elSize, elType, RNS, RSA } from "../../utils/type.ts"
import { UploadProps } from "../hUpload/index.vue"
import { PreviewProps } from "../hPreview/index.vue"

export interface FormOptionItem extends UploadProps, PreviewProps {
  prop?: string
  ref?: string
  type: string
  label?: string
  placeholder?: string
  content?: string
  clearable?: boolean
  disabled?: boolean
  show?: boolean
  readonly?: boolean
  key?: string | number | boolean
  labelWidth?: string | number
  itemWidth?: string | number
  size?: elSize
  // button--------------------------------------------------------------------
  btnType?: elType
  click?: (o?: RSA) => void
  change?: (o?: RSA) => void
  enter?: (o?: RSA) => void
  mouseover?: (o?: RSA) => void
  btnClassName?: string
  plain?: boolean
  // tag-----------------------------------------------------------------------
  tagType?: elType
  tagClassName?: string
  // input---------------------------------------------------------------------
  maxlength?: string | number
  minlength?: string | number
  showPassword?: boolean
  showWordLimit?: boolean
  controls?: boolean
  stepStrictly?: boolean
  prepend?: boolean
  append?: boolean
  prefix?: boolean
  suffix?: boolean
  inputType?: "text" | "textarea"
  controlsPosition?: "" | "right"
  max?: number
  min?: number
  precision?: number
  row?: number
  step?: number
  // select--------------------------------------------------------------------
  option?: RSA[]
  optionKey?: string
  optionVal?: string
  valueKey?: string
  allowCreate?: boolean
  multiple?: boolean
  // cascader------------------------------------------------------------------
  collapseTags?: boolean
  debounce?: number
  props?: RSA
  separator?: string
  // 提示信息-------------------------------------------------------------------
  hintLeft?: string
  hintRight?: string
  hintIcon?: string
  // time date picker----------------------------------------------------------
  editable?: boolean
  format?: string
  valueFormat?: string
  startPlaceholder?: string
  endPlaceholder?: string
  rangeSeparator?: string
  disabledDate: () => void
  defaultTime?: Date | [Date, Date]
  // switch -------------------------------------------------------------------
  loading?: boolean
  width?: number | string
  activeColor?: string
  inactiveColor?: string
  inactiveText?: string
  inactiveValue?: boolean | number | string
  activeText?: string
  activeValue?: boolean | number | string
  activeIcon?: string
  inactiveIcon?: string
  // text ---------------------------------------------------------------------
  textClassName?: string
  color?: string
  textType?: elType
  truncated?: boolean
  tag?: string
  // radio --------------------------------------------------------------------
  fill?: string
  textColor?: string
  button?: boolean
  border?: boolean
  // slider--------------------------------------------------------------------
  formatTooltip: () => void
  marks?: RSA
  range?: boolean
  showInput?: boolean
  showInputControls?: boolean
  showStops?: boolean
  tooltip?: boolean
  vertical?: boolean
  tooltipClass?: string
  // rate 评分------------------------------------------------------------------
  allowHalf?: boolean
  top?: number
  colors?: string[] | RNS
  disabledVoidColor?: string
  disabledVoidIcon?: string
  highThreshold?: number
  lowThreshold?: number
  icon?: string
  showScore?: boolean
  showText?: boolean
  texts?: string[]
  voidColor?: string
  voidIcon?: string
  // divider ------------------------------------------------------------------
  dividerLeftIcon?: string
  dividerRightIcon?: string
  contentPosition?: "left" | "right" | "center"
}
