// otb-module-plus/src/utils/tool/check.ts
// 数据校验类型
import { RSA, RSS } from "../type"

export const regular: RSA = {
  link: /^(https?:|mailto:|tel:)/,
  number: /\D/g,
  decimals: /[^\\d.]/g,
  phone: /^1[3|4|5|6|7|8|9][0-9]\d{8}$/,
  bank: /^([1-9]{1})(\d{15}|\d{16}|\d{18})$/,
  idCard:
    /^\d{6}(18|19|20)?\d{2}(0[1-9]|1[012])(0[1-9]|[12]\d|3[01])\d{3}(\d|X)$/i
}

// 校验身份证
export const idCardVerify = (code: string) => {
  const city: RSS = {
    "11": "北京",
    "12": "天津",
    "13": "河北",
    "14": "山西",
    "15": "内蒙古",
    "21": "辽宁",
    "22": "吉林",
    "23": "黑龙江 ",
    "31": "上海",
    "32": "江苏",
    "33": "浙江",
    "34": "安徽",
    "35": "福建",
    "36": "江西",
    "37": "山东",
    "41": "河南",
    "42": "湖北 ",
    "43": "湖南",
    "44": "广东",
    "45": "广西",
    "46": "海南",
    "50": "重庆",
    "51": "四川",
    "52": "贵州",
    "53": "云南",
    "54": "西藏 ",
    "61": "陕西",
    "62": "甘肃",
    "63": "青海",
    "64": "宁夏",
    "65": "新疆",
    "71": "台湾",
    "81": "香港",
    "82": "澳门",
    "91": "国外 "
  }

  if (!code || !regular.idCard.test(code)) return "身份证号格式错误"
  if (!city[code.substr(0, 2)]) return "身份证输入错误"

  // 18位身份证需要验证最后一位校验位
  if (code.length === 18) {
    const arr = code.split(",")
    // ∑(ai×Wi)(mod 11)
    // 加权因子
    const factor = [7, 9, 10, 5, 8, 4, 2, 1, 6, 3, 7, 9, 10, 5, 8, 4, 2]
    // 校验位
    const parity = ["1", "0", "X", "9", "8", "7", "6", "5", "4", "3", "2"]
    let sum = 0
    let ai = 0
    let wi = 0
    for (let i = 0; i < 17; i += 1) {
      ai = Number(arr[i])
      wi = factor[i]
      sum += ai * wi
    }
    if (parity[sum % 11] !== arr[17]) return "身份证输入错误"
  }

  return true
}

// 排除校验
export const exclude = (value: any, type: string) => {
  const reg = regular[type]
  if (!reg) return "不存在此校验类型"
  return value.replace(reg, "")
}

// 判断校验
export const validator = (value: any, type: string) => {
  const reg = regular[type]
  if (!reg) return "不存在此校验类型"
  if (type === "idCard") return idCardVerify(value)
  return reg.test(value)
}
