// otb-module-plus/src/utils/tool/data.ts
import { order, RSA, sortTarget } from "../type"

// 判断类型
export const type = (value: unknown) => {
  const result = Object.prototype.toString.call(value)
  if (result === "[object Array]") return "Array"
  if (result === "[object Object]") return "Object"
  if (result === "[object String]") return "String"
  if (result === "[object Number]") return "Number"
  if (result === "[object Undefined]") return "Undefined"
  if (result === "[object Boolean]") return "Boolean"
  if (result === "[object Function]") return "Function"
  if (result === "[object Date]") return "Date"
}

// 递归获取所需对象
export const getObject = (
  target: RSA[],
  id: number | string,
  props = {
    id: "id",
    children: "children"
  }
) => {
  let result
  const setObj = (item: RSA) => {
    if (item[props.id] === id) {
      result = item
      return
    }
    if (item[props.children] && type(item[props.children]) === "Array") {
      for (const childItem of item[props.children]) setObj(childItem)
    }
  }
  for (const item of target) setObj(item)

  return result
}

// 排序 数字数组 对象数组 字符串数组
export const sort = (target: sortTarget, order: order, key: string) => {
  return target.sort((p1, p2) => {
    if (type(target[0]) === "Number") {
      const val1 = p1 as number
      const val2 = p2 as number
      return order === "asc" ? val1 - val2 : val2 - val1
    }

    if (type(target[0]) === "String") {
      const val1 = (p1 as string).toUpperCase()
      const val2 = (p2 as string).toUpperCase()
      if (val1 < val2) return order === "asc" ? -1 : 1
      if (val1 > val2) return order === "asc" ? 1 : -1
      return 0
    }

    if (type(target[0]) === "Object") {
      const item1 = p1 as RSA
      const item2 = p2 as RSA

      if (type(item1[key]) === "Number") {
        const val1 = item1[key] as number
        const val2 = item2[key] as number
        return order === "asc" ? val1 - val2 : val2 - val1
      }

      if (type(item1[key]) === "String") {
        const val1 = (item1[key] as string).toUpperCase()
        const val2 = (item2[key] as string).toUpperCase()
        if (val1 < val2) return order === "asc" ? -1 : 1
        if (val1 > val2) return order === "asc" ? 1 : -1
        return 0
      }
    }

    return 0
  })
}

// tree按某个字段排序
export const treeSort = (
  target: RSA[],
  order: order,
  props = {
    sortKey: "sort",
    children: "children"
  }
) => {
  const setObj = (item: RSA) => {
    if (item[props.children] && type(item[props.children]) === "Array") {
      sort(item[props.children], order, props.sortKey)
      for (const childItem of item[props.children]) setObj(childItem)
    }
  }

  sort(target, order, props.sortKey)
  for (const item of target) setObj(item)

  return target
}

// 将一个数组分割成多个小数组
export const getSliceArray = (array: unknown[], subArrayLength: number) => {
  let index = 0
  const newArray = []
  while (index < array.length) {
    newArray.push(array.slice(index, (index += subArrayLength)))
  }
  return newArray
}
