// otb-module-plus/src/utils/tool/date.ts
import { dayjs } from "element-plus"
import { DateType } from "../type"

class Day {
  format: string

  constructor(format: string) {
    this.format = format
  }

  type(date?: DateType) {
    return dayjs(date).format(this.format)
  }
}

// 日期格式化
export const day = {
  datetime: (d?: DateType) => new Day("YYYY-MM-DD HH:mm:ss").type(d),
  date: (d?: DateType) => new Day("YYYY-MM-DD").type(d),
  time: (d?: DateType) => new Day("HH:mm:ss").type(d),
  year: (d?: DateType) => new Day("YYYY").type(d)
}
// 获取某月天数方法
export const monthNumber = (date: DateType) => {
  const year = new Date(date).getFullYear()
  const month = new Date(date).getMonth() + 1

  return new Date(year, month, 0).getDate()
}
// 获取某年天数方法
export const yearNumber = (n = 0) => {
  n = Math.floor(Math.abs(n))
  n = new Date().getFullYear() - n

  let number = 0
  for (let i = 1; i <= 12; i += 1) {
    number += new Date(n, i, 0).getDate()
  }

  return number
}
// 多久以前
export const howLongAgo = (d = 0, m = 0, y = 0) => {
  d = Math.floor(Math.abs(d))
  m = Math.floor(Math.abs(m))
  y = Math.floor(Math.abs(y))

  const one = 24 * 60 * 60 * 1000
  const nowTimestamp = new Date().getTime()

  // 按月算多少天
  let month = 0
  if (m > 0) {
    for (let i = 1; i <= m; i += 1) {
      month += monthNumber(i)
    }
  }
  // 按年算多少天
  let year = 0
  if (y > 0) {
    for (let i = 1; i <= y; i += 1) {
      year += yearNumber(i)
    }
  }

  const timestamp = nowTimestamp - (d + month + year) * one

  return day.datetime(timestamp)
}
// 时间戳
export const timestamp = (date: DateType) => {
  return new Date(date).getTime()
}
// 时间差
export const timeDifference = (begin: DateType, end: DateType) => {
  const difference = timestamp(end) - timestamp(begin)
  return Math.abs(difference / (24 * 60 * 60 * 1000)) + 1
}
