// otb-module-plus/src/utils/tool/math.ts
class HMath {
  type: string

  constructor(type: any) {
    this.type = type
  }

  math(arg1: number, arg2: number) {
    const r1 = arg1.toString().split(".")[1].length || 0
    const r2 = arg2.toString().split(".")[1].length || 0
    const t1 = Number(arg1.toString().replace(".", ""))
    const t2 = Number(arg2.toString().replace(".", ""))
    if (this.type === "add") {
      const m = 10 ** Math.max(r1, r2)
      return (arg1 * m + arg2 * m) / m
    }
    if (this.type === "sub") {
      const n = Math.max(r1, r2)
      const m = 10 ** n
      return Number(((arg1 * m - arg2 * m) / m).toFixed(n))
    }
    if (this.type === "mul") {
      return (t1 * t2) / 10 ** (r1 + r2)
    }
    if (this.type === "div") {
      return (t1 / t2) * 10 ** (r1 - r2)
    }
  }
}

// 加
export const add = (arg1: number, arg2: number) => {
  return new HMath("add").math(arg1, arg2)
}
// 减
export const sub = (arg1: number, arg2: number) => {
  return new HMath("sub").math(arg1, arg2)
}
// 乘
export const mul = (arg1: number, arg2: number) => {
  return new HMath("mul").math(arg1, arg2)
}
// 除
export const div = (arg1: number, arg2: number) => {
  return new HMath("div").math(arg1, arg2)
}
// 三位分节法（千分位加逗号）
export const toThousandsString = (arg: number) => {
  const str = arg.toString()
  const reg = str.indexOf(".") ? /(\d)(?=(\d{3})+\.)/g : /(\d)(?=(?:\d{3})+$)/g
  return str.replace(reg, "$1,")
}
