// otb-module-plus/src/components/hCrud/interface.ts
import { FormOptionItem } from "../hForm/interface.ts"
import { FormProps } from "../hForm/index.vue"
import { elSize, elType, RSA } from "../../utils/type.ts"

export interface CRUDOptionColumn extends FormOptionItem {
  preview?: boolean
  previewHeight?: string | number
  previewWidth?: string | number
  button?: boolean
  date?: boolean
  datetime?: boolean
  unit?: string
  className?: string
  colType?: "selection" | "index" | "expand" | "operation"
  align?: "center" | "right"
  fixed?: true | "left" | "right"
  width?: number | string
  formWidth?: number | string
  useSlot?: boolean
  useForm?: boolean
  formOption?: FormOptionItem[]
}

export interface CRUDOption {
  fit?: boolean
  border?: boolean
  summary?: boolean
  currentRow?: boolean
  tableSelection?: boolean
  tableSelectionFixed?: boolean
  tableIndex?: boolean
  operationShow?: boolean
  tableIndexFixed?: boolean
  height?: string | number
  maxHeight?: string | number
  tableSelectionWidth?: string | number
  tableIndexWidth?: string | number
  operationWidth?: string | number
  LoadingText?: string
  labelWidth?: string | number
  formWidth?: string | number
  itemWidth?: string | number
  className?: string
  loadingIcon?: string
  tableIndexLabel?: string
  operationLabel?: string
  column: CRUDOptionColumn[]
  size?: elSize
  summaryList?: string[]
  summaryFields?: RSA
  viewBtn?: FormOptionItem
  updBtn?: FormOptionItem
  delBtn?: FormOptionItem
  customBtn?: FormOptionItem[]
}

export interface CRUDOperation extends FormProps {
  addBtn?: FormOptionItem
  exportBtn?: FormOptionItem
  customBtn?: FormOptionItem[]
}

export interface CRUDDialog extends FormProps {
  title?: string
  width?: number | string
  confirmBtn?: FormOptionItem
  confirmBtnSize?: elSize
  confirmBtnType?: elType
  confirmBtnName?: string
  cancelBtn?: FormOptionItem
  cancelBtnSize?: elSize
  cancelBtnType?: elType
  cancelBtnName?: string
}
